backend/src/test/kotlin/blue/mild/covid/vaxx/routes/NurseRoutesTest.kt

```diff
@@ -24,19 +27,34 @@ import kotlin.test.assertTrue
 
 class NurseRoutesTest : ServerTestBase() {
     @Test
-    @Disabled
     fun `should respond with all nurses`() = withTestApplication {
-        TODO("implement this test for file NurseRoutes.kt")
-        // verify two cases:
         // 1. when user submits correct credentials, server should respond with all nurses in the database
-        // 2. in case of invalid credentials, the server should respond with status 401
+        val validLogin = LoginDtoIn(
+            credentials = CredentialsDtoIn(DatabaseData.admin.email, DatabaseData.admin.password),
+            nurseId = null,
+            vaccineSerialNumber = "",
+            vaccineExpiration = LocalDate.now()
+        )
 
-        // very similar test is for example UserRoutesTest, where we try to log in and then check
-        // what status code the server returned
+        handleRequest(HttpMethod.Post, Routes.registeredUserLogin) {
+            jsonBody(validLogin)
+        }.run {
+            expectStatus(HttpStatusCode.OK)
+            val nurseRepository by closestDI().instance<NurseRepository>()
+            runBlocking { nurseRepository.getAll() }
```

Comment:
Here you don't check the values returned by the server (from the http request) but values returned by the database.

You should check that nurses returned by the database `runBlocking { nurseRepository.getAll() }` are the same as the ones that were returned by the server in the http request.

You can see how to retrieve body of the request [here](https://github.com/mild-blue/covid-vaxx/blob/master/backend/src/test/kotlin/blue/mild/covid/vaxx/routes/UserRoutesTest.kt#L70). The correct type that needs to be retrieved is then in route itself [here](https://github.com/mild-blue/covid-vaxx/blob/master/backend/src/main/kotlin/blue/mild/covid/vaxx/routes/UserRoute.kt#L38)